### PR TITLE
Add CppADCodeGen

### PR DESCRIPTION
**Pull request by jcarpent**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jcarpent** [comment]
@traversaro Do you agree to be maintainer of this package?

**Comment by traversaro** [comment]
> 
> 
> @traversaro Do you agree to be maintainer of this package?

Ok!

**Comment by jcarpent** [comment]
@conda-forge/help-c-cpp This package is ready for review. Thanks a lot for your help and time.

**Comment by jcarpent** [comment]
@conda-forge/help-c-cpp a kind reminder.

Best,
Justin

File: recipes/cppadcodegen/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+test:
+  commands:
+    - test -d ${PREFIX}/include/cppad/cg  # [unix]
+    - test -f ${PREFIX}/share/pkgconfig/cppadcg.pc  # [unix]
+    - if not exist %PREFIX%\\Library\\include\\cppad\cg exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\share\\pkgconfig\\cppadcg.pc exit 1  # [win]
+
+about:
+  home: https://github.com/joaoleal/CppADCodeGen
+  license: GPL-3.0-only AND EPL-2.0
```
Comment (traversaro) [inline recipes/cppadcodegen/meta.yaml:36]: From https://github.com/joaoleal/CppADCodeGen#license I guess that you can choose either license, so perhaps the correct license is `OR` .
```suggestion
  license: GPL-3.0-only OR EPL-2.0 
```

Comment (jcarpent) [reply recipes/cppadcodegen/meta.yaml]: Thanks

**Comment by jcarpent** [comment]
@wolfv Would you have time to review this PR? Thanks in advance.
Best,

Justin

**Comment by wolfv** [comment]
Awesome!

**Comment by jcarpent** [comment]
Thanks a lot @wolfv. Very efficient.

**Comment by wolfv** [comment]
Sorry for the wait

**Pull request merged**